src/tool/cppwinrt/strings/base_collections_base.h

```diff
@@ -103,17 +103,35 @@ namespace winrt
 
             uint32_t GetMany(array_view<T> values)
             {
-                uint32_t const actual = (std::min)(static_cast<uint32_t>(std::distance(m_current, m_end)), values.size());
-                m_owner->copy_n(m_current, actual, values.begin());
-                std::advance(m_current, actual);
-                return actual;
+                if constexpr (std::is_same_v<decltype(typename std::iterator_traits<iterator_type>::iterator_category()), std::random_access_iterator_tag>)
```

Comment:
`std::is_same` is the wrong way to light up behavior for random access iterators, as it will improperly reject C++20's `contiguous_iterator_tag`. The iterator tags have an inheritance chain, basically representing the strength of the iterator guarantees - `continuous_iterator_tag` inherits from `random_access_iterator_tag`, which inherits from `bidirectional_iterator_tag`, and so on.

What you really need to do here is use overload dispatch to specialize the behavior based on iterator "strength", as in the example at https://en.cppreference.com/w/cpp/iterator/iterator_tags.

Reply:
Ah good idea. I'll do that.